pal/inc/pal.h

```diff
@@ -37,6 +37,8 @@ Module Name:
 #ifndef __PAL_H__
 #define __PAL_H__
 
+#include <assert.h>
```

Comment:
I don't quite understand why this line causes OSX build to fail with `use of undeclared identifier 'llabs'`:

```
[  0%] Building CXX object pal/src/CMakeFiles/Chakra.Pal.dir/cruntime/file.cpp.o
In file included from /Users/runner/runners/2.166.3/work/1/s/pal/src/cruntime/file.cpp:23:
In file included from /Users/runner/runners/2.166.3/work/1/s/pal/src/include/pal/palinternal.h:323:
In file included from /Users/runner/runners/2.166.3/work/1/s/pal/inc/pal.h:40:
In file included from /Applications/Xcode_11.3.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/assert.h:44:
/Applications/Xcode_11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/stdlib.h:113:81: error: use of undeclared identifier 'llabs'; did you mean 'labs'?
```
https://chakrateam.visualstudio.com/ChakraCore%20CI/_build/results?buildId=47202

This sounds like mismatch between `stdlib.h` and `assert.h`. @rhuanjl, @zenparsing have you seen something like this before?

Reply:
Hmm... So the assert.h it's finding is the "standard C" one, whereas the stdlib.h it's error-ing in is the C++ one - the C stdlib.h defines "llabs" and the C++ one undefs it and re-defines it. 

However in the C header the definition of llabs is guarded by `#if !__DARWIN_NO_LONG_LONG` BUT in the C++ header the undef is guarded by `#ifndef _LIBCPP_HAS_NO_LONG_LONG` so if __DARWIN_NO_LONG_LONG is true BUT _LIBCPP_HAS_NO_LONG_LONG is not defined you'd hit this error.

To be honest I've never come across these macros before.

Reply:
As an alternative could these problems be a sequencing issue with all of the #undefs in palinternal.h?

It undefs most of the standard cruntime so easy to hit things being undefined based on the ordering of includes before/after palinternal.

Reply:
Had another look - attempting to include extra standard headers in the middle of PAL is just a nightmare...

However can fix the warnings by using `#define static_assert _Static_assert` at the top of Pal.h instead of including assert.h

Reply:
That might be a better way to do it.

As an aside, we appear to be using "magic" compiler symbols in quite a few places (this change touches one more place like that). It would be hard to rid PAL of those, but in other parts we should strive to detect what headers are present and then use standard symbols.